lib/block-supports/duotone.php

```diff
@@ -450,3 +471,4 @@ function gutenberg_render_duotone_support( $block_content, $block ) {
 add_action( 'wp_enqueue_scripts', array( 'WP_Duotone_Gutenberg', 'output_global_styles' ), 11 );
 add_action( 'wp_footer', array( 'WP_Duotone_Gutenberg', 'output_footer_assets' ), 10 );
 add_filter( 'block_editor_settings_all', array( 'WP_Duotone_Gutenberg', 'add_editor_settings' ), 10 );
+add_filter( 'block_type_metadata_settings', 'gutenberg_migrate_experimental_duotone_support_flag', 10, 2 );
```

Comment:
The only problem is that there's no longer any way of knowing if you have experimental support or not. It's causing me problems because the selector that we choose is different depending on whether or not it's experimental.

```js
// May be true even if `filter.duotone` isn't set.
const duotoneSupport = getBlockSupport(
	blockType,
	'filter.duotone',
	false
);

// May be set even if `filter.duotone` is also set.
const experimentalDuotone = getBlockSupport(
	blockType,
	'color.__experimentalDuotone',
	false
);

// Will always produce a selector even if `filter.duotone` isn't set.
const duotoneSelector = getBlockCSSSelector(
	blockType,
	'filter',
	{ fallback: true }
);
```

Reply:
I'm not sure I follow.

In the filter, we were only setting `filter.duotone` if it hasn't already been set and the `color.experimentalDuotone` has been.

A block author shouldn't be using both properties in their raw block.json so if we have `color.experimentalDuotone` set at all, then it should be a case from when duotone was experimental.

- In your first example, the check for `filter.duotone` tells us if there is support.
- The second example if the default is removed or changed to `null` would tell us if the duotone support was the experimental version.
- The third snippet with fallback true would always produce a selector regardless of any support check. We would guard such a call with a check that the `color.experimentalDuotone` support flag wasn't set.

Reply:
My thinking was that by leaving the `__experimentalI can see from this discussion where I missed adding the guard check to the duotone.js HoC.